tools/sdk-sync/src/sync/gen.rs

```diff
@@ -114,6 +114,9 @@ impl DefaultSdkGenerator {
         let mut command = Command::new("./gradlew");
         command.arg("--no-daemon"); // Don't let Gradle continue running after the build
         command.arg("--info"); // Increase logging verbosity for failure debugging
+
+        // Disable Smithy's codegen parallelism in favor of sdk-sync parallelism
+        command.arg("-Djava.util.concurrent.ForkJoinPool.common.parallelism=1");
```

Comment:
lets make this configurable? I think we _may_ want to have at least some parallelism here? (eg. this probably massively reduces the memory burden)

Whatever we can do to make this easy to tweak in the build job without redeploying the image would be 💯